src/main/world/turbo.gd

```diff
@@ -216,26 +229,50 @@ func jump() -> void:
 
 
 """
-When Turbo attempts to jump onto a narrow surface, she slides off unless she lands very squarely in the middle of it.
+When Turbo attempts to jump onto a narrow surface (such as someone's head) she slips off unless she lands very
+squarely in the middle of it.
+
+This function triggers the slipping physics, and returns 'true' if Turbo is slipping from a narrow surface.
 """
-func _slide_from_narrow_surfaces(delta: float) -> void:
-	if is_on_floor():
-		var just_slipped = false
-		if get_slide_count() > 0:
-			for i in get_slide_count():
-				var collision = get_slide_collision(i)
-				if collision.collider.get("foothold_radius") and translation.y > collision.collider.translation.y + 0.2:
-					# Turbo is standing on something she might slip from
-					var slip_velocity: Vector3 = translation - collision.collider.translation
-					slip_velocity.y = 0
-					if slip_velocity.length() > collision.collider.foothold_radius:
-						# Turbo is too far from the center, and should slip
-						_slipping = true
-						just_slipped = true
-						accelerate_player_xy(delta, Vector2(slip_velocity.x, slip_velocity.z), MAX_SLIP_ACCELERATION, MAX_SLIP_SPEED)
-						$CoyoteTimer.stop()
-						$JumpBuffer.stop()
-		
-		if _slipping and not just_slipped:
-			# Turbo was slipping, but they've landed on something non-slippery.
-			_slipping = false
+func _slip_from_narrow_surfaces(delta: float) -> bool:
+	var just_slipped = false
+	if get_slide_count() > 0:
+		for i in get_slide_count():
+			var collision = get_slide_collision(i)
+			if collision.collider.get("foothold_radius") and translation.y > collision.collider.translation.y + 0.2:
+				# Turbo is standing on something she might slip from
+				var slip_velocity: Vector3 = translation - collision.collider.translation
+				slip_velocity.y = 0
+				if slip_velocity.length() > collision.collider.foothold_radius:
+					# Turbo is too far from the center, and should slip
+					just_slipped = true
+					accelerate_player_xy(delta, Vector2(slip_velocity.x, slip_velocity.z), MAX_SLIP_ACCELERATION, MAX_NARROW_SLIP_SPEED)
+					$CoyoteTimer.stop()
+					$JumpBuffer.stop()
```

Comment:
This `CoyoteTimer.stop JumpBuffer.stop` code should probably be pulled out, as it likely applies to slipping from ledges as well. Both or neither.

Reply:
Removed `CoyoteTimer.stop` `JumpBuffer.stop` code -- i've added an item to the backlog to consider what the CoyoteTimer should do now. Maybe the player needs to be able to jump while slipping in some cases.